### ISSUE TITLE
mpi_test_suite -t get_fence -d MPI_TYPE_MIX,MPI_TYPE_MIX_ARRAY with tcp on nodes=16,ppn=28

### ISSUE THREAD

**Issue by dmitrygx** [openucx/ucx#3758]
```
mpirun --mca pml ucx --mca btl ^vader,tcp,openib -x xUCX_LOG_LEVEL=debug -x UCX_NET_DEVICES=p2p2 -x UCX_TLS=tcp --oversubscribe --map-by node -x LD_LIBRARY_PATH=/auto/mtrswgwork/dmitrygla/install/lib/:$LD_LIBRARY_PATH --oversubscribe --map-by node ~/work_auto/ompi-tests/mpi_test_suite/mpi_test_suite -t get_fence -d MPI_TYPE_MIX,MPI_TYPE_MIX_ARRAY
```

hangs on 16 nodes with ppn=28 on master and 1.6.x (other wasn't checked)

1.6.x reports:
```
[1561202028.249446] [jazz20:57994:0]           sock.c:208  UCX  ERROR recv(fd=1218 data=0x7ffd0e9c71f0 length=4) failed: Connection reset by peer
[1561202028.273034] [jazz20:57985:0]           sock.c:208  UCX  ERROR recv(fd=1176 data=0x7fffb76077e0 length=4) failed: Connection reset by peer
[1561202028.300348] [jazz20:57977:0]           sock.c:208  UCX  ERROR recv(fd=1203 data=0x7ffd4d682620 length=4) failed: Connection reset by peer
[1561202028.311638] [jazz20:58002:0]           sock.c:208  UCX  ERROR recv(fd=1168 data=0x7ffd0a5ef6e0 length=4) failed: Connection reset by peer
[1561202028.314436] [jazz20:57985:0]           sock.c:208  UCX  ERROR recv(fd=1160 data=0x7fffb76077e0 length=4) failed: Connection reset by peer
[1561202028.314449] [jazz20:57985:0]           sock.c:208  UCX  ERROR recv(fd=1208 data=0x7fffb76077e0 length=4) failed: Connection reset by peer
[1561202028.404437] [jazz20:57964:0]           sock.c:208  UCX  ERROR recv(fd=1135 data=0x7ffe168a01e0 length=4) failed: Connection reset by peer
[1561202028.404465] [jazz20:57964:0]           sock.c:208  UCX  ERROR recv(fd=1171 data=0x7ffe168a01e0 length=4) failed: Connection reset by peer
[1561202028.401011] [jazz20:57967:0]           sock.c:208  UCX  ERROR recv(fd=1177 data=0x7ffcdef33860 length=4) failed: Connection reset by peer
[1561202028.401057] [jazz20:57967:0]           sock.c:208  UCX  ERROR recv(fd=1219 data=0x7ffcdef33860 length=4) failed: Connection reset by peer
[1561202028.418356] [jazz20:57965:0]           sock.c:208  UCX  ERROR recv(fd=1155 data=0x7fffd7a063c0 length=4) failed: Connection reset by peer
[1561202029.247962] [jazz20:57989:0]           sock.c:208  UCX  ERROR recv(fd=1170 data=0x7ffc829a8220 length=4) failed: Connection reset by peer
[1561202029.248026] [jazz20:57989:0]           sock.c:208  UCX  ERROR recv(fd=1186 data=0x7ffc829a8220 length=4) failed: Connection reset by peer
[1561202029.600810] [jazz20:57978:0]           sock.c:208  UCX  ERROR recv(fd=1164 data=0x7ffc08bf4d90 length=4) failed: Connection reset by peer
[1561202030.555237] [jazz20:57987:0]           sock.c:208  UCX  ERROR recv(fd=1198 data=0x7ffd39b50360 length=4) failed: Connection reset by peer
[1561202031.285472] [jazz20:57978:0]           sock.c:208  UCX  ERROR recv(fd=1171 data=0x7ffc08bf4d90 length=4) failed: Connection reset by peer
[1561202031.356378] [jazz20:57967:0]           sock.c:208  UCX  ERROR recv(fd=1171 data=0x7ffcdef33860 length=4) failed: Connection reset by peer
...
```

Separately those tests work fine

**Comment by yosefe** [comment]
@dmitrygladkov does it fail only with TCP or also with IB?

**Comment by dmitrygladkov** [comment]
> does it fail only with TCP or also with IB?

It works fine with IB transports
Only UCX_TLS=tcp fails

**Comment by dmitrygx** [comment]
number of ranks = 1024, number of nodes = 32
fails with:
```
(Rank:0) tst_test_array[0]:get_fence
One-sided tests get_fence (49/1), comm MPI_COMM_WORLD (1/14), type MPI_TYPE_MIX (27/1)
[1567587583.918304] [clx-orion-053:11706:0]           sock.c:228  UCX  ERROR recv(fd=1377) failed: 104
```

**Comment by dmitrygx** [comment]
```
dmitrygla@mngx-orion-01 ~
$cat ucx_log_file_clx-hercules-024_6442 | grep -ni "172.17.11.19:57150"
631:[1568200100.251316] [clx-hercules-024:6442 :0]           sock.c:142  UCX  DEBUG connect(fd=54, src_addr=172.17.11.24:35714 dest_addr=172.17.11.19:57150): Operation now in progress
632:[1568200100.251319] [clx-hercules-024:6442 :0]         tcp_cm.c:90   UCX  DEBUG tcp_ep 0x240ea20: CLOSED -> CONNECTING for the [172.17.11.24:49908]<->[172.17.11.19:57150] connection [-:-]
8168:[1568200100.299858] [clx-hercules-024:6442 :0]         tcp_cm.c:156  UCX  TRACE tcp_ep 0x240ea20: UCT_TCP_CM_CONN_REQ sent to 172.17.11.19:57150
8169:[1568200100.299861] [clx-hercules-024:6442 :0]         tcp_cm.c:90   UCX  DEBUG tcp_ep 0x240ea20: CONNECTING -> WAITING_ACK for the [172.17.11.24:49908]<->[172.17.11.19:57150] connection [Tx:-]
13187:[1568200100.330440] [clx-hercules-024:6442 :0]         tcp_cm.c:156  UCX  TRACE tcp_ep 0x7fffdc002870: UCT_TCP_CM_CONN_REQ received from 172.17.11.19:57150
13188:[1568200100.330443] [clx-hercules-024:6442 :0]         tcp_cm.c:90   UCX  DEBUG tcp_ep 0x7fffdc002870: ACCEPTING -> CONNECTING for the [172.17.11.24:49908]<->[172.17.11.19:57150] connection [-:-]
13198:[1568200100.330468] [clx-hercules-024:6442 :0]         tcp_cm.c:90   UCX  DEBUG tcp_ep 0x7fffdc002870: CONNECTING -> CLOSED for the [172.17.11.24:49908]<->[172.17.11.19:57150] connection [-:-]
13200:[1568200100.330476] [clx-hercules-024:6442 :0]         tcp_cm.c:156  UCX  TRACE tcp_ep 0x240ea20: UCT_TCP_CM_CONN_WAIT_REQ | UCT_TCP_CM_CONN_ACK sent to 172.17.11.19:57150
13202:[1568200100.330484] [clx-hercules-024:6442 :0]         tcp_cm.c:90   UCX  DEBUG tcp_ep 0x240ea20: WAITING_ACK -> CONNECTED for the [172.17.11.24:49908]<->[172.17.11.19:57150] connection [Tx:Rx]

dmitrygla@mngx-orion-01 ~
$cat ucx_log_file_clx-hercules-019_22373 | grep -ni "accepted connection from 172.17.11.24:35714"
dmitrygla@mngx-orion-01 ~
$cat ucx_log_file_clx-hercules-019_22373 | grep -ni "accepted connection from 172.17.11.24:"
7698:[1568200100.320669] [clx-hercules-019:22373:0]         tcp_cm.c:611  UCX  DEBUG tcp_iface 0x23dd840: accepted connection from 172.17.11.24:35596 on 172.17.11.19:57150 to tcp_ep 0x242b1d0 (fd 375)
7706:[1568200100.320696] [clx-hercules-019:22373:0]         tcp_cm.c:611  UCX  DEBUG tcp_iface 0x23dd840: accepted connection from 172.17.11.24:35566 on 172.17.11.19:57150 to tcp_ep 0x242b370 (fd 377)
7962:[1568200100.321614] [clx-hercules-019:22373:0]         tcp_cm.c:611  UCX  DEBUG tcp_iface 0x23dd840: accepted connection from 172.17.11.24:35628 on 172.17.11.19:57150 to tcp_ep 0x242e770 (fd 441)
7986:[1568200100.321690] [clx-hercules-019:22373:0]         tcp_cm.c:611  UCX  DEBUG tcp_iface 0x23dd840: accepted connection from 172.17.11.24:35692 on 172.17.11.19:57150 to tcp_ep 0x242ec50 (fd 447)
11359:[1568200100.341293] [clx-hercules-019:22373:1]         tcp_cm.c:611  UCX  DEBUG tcp_iface 0x23dd840: accepted connection from 172.17.11.24:35718 on 172.17.11.19:57150 to tcp_ep 0x7fffdc0019d0 (fd 143)
11403:[1568200100.341418] [clx-hercules-019:22373:1]         tcp_cm.c:611  UCX  DEBUG tcp_iface 0x23dd840: accepted connection from 172.17.11.24:35636 on 172.17.11.19:57150 to tcp_ep 0x7fffdc0022c0 (fd 204)
11411:[1568200100.341439] [clx-hercules-019:22373:1]         tcp_cm.c:611  UCX  DEBUG tcp_iface 0x23dd840: accepted connection from 172.17.11.24:35666 on 172.17.11.19:57150 to tcp_ep 0x7fffdc002460 (fd 207)
11427:[1568200100.341485] [clx-hercules-019:22373:1]         tcp_cm.c:611  UCX  DEBUG tcp_iface 0x23dd840: accepted connection from 172.17.11.24:35728 on 172.17.11.19:57150 to tcp_ep 0x7fffdc0027a0 (fd 228)
19234:[1568200100.496078] [clx-hercules-019:22373:1]         tcp_cm.c:611  UCX  DEBUG tcp_iface 0x23dd840: accepted connection from 172.17.11.24:35698 on 172.17.11.19:57150 to tcp_ep 0x7fffdc000da0 (fd 50)

dmitrygla@mngx-orion-01 ~
$cat ucx_log_file_clx-hercules-019_22373 | grep -ni "accepted connection from 172.17.11.24:35714" | wc -l
0
```
